doc/admin-guide/files/remap.config.en.rst

```diff
@@ -518,20 +514,95 @@ mapping rules. (It is activated before any mappings and is never
 deactivated.) The filter `local_only` will only be applied to the
 second mapping.
 
-Implict IPAllow filter
-======================
+Special Filter and ip_allow Named Filter
+----------------------------------------
+
+If :file:`ip_allow.yaml` has a "deny all" filter, it is treated as a special filter that is applied before remapping for
+optimizaion. To control this for specific remap rules, a named filter called `ip_allow` is pre-defined. This named filter is
+activated implicitly in default. To stop applying the special rule, disable the `ip_allow` filter as shown below.
+
+::
 
-To allow control of :ref:`IP Allow<ip-allow>` it is treated as an implicitly active and named
-filter. When this filter is active IP Allow checks are done before remap. To prevent this for
-specific remap rules, this filter, named "ip_allow", must be disabled. The common way of doing this
-would be ::
+   # ip_allow.yaml
+   ip_allow:
+      - apply: in
+        ip_addrs: 198.51.100.0/24
+        action: deny
+        method: ALL
 
+   # remap.config
    .deactivatefilter ip_allow
    map ...
    map ...
    .activatefilter ip_allow
 
-Note this entirely disables IP Allow checks for those remap rules.
+Note this entirely disables :file:`ip_allow.yaml` checks for those remap rules.
+
+Evaluation Order and Matching Policy
+------------------------------------
+
+ATS evaluates multiple ACL filters in the following order:
+
+1. Special "deny all" filter in :file:`ip_allow.yaml`
+2. In-line Filter in :file:`remap.config`
+3. Named Filter in :file:`remap.config`
+4. Filters in :file:`ip_allow.yaml`
+
+When an ACL filter is found, ATS stops processing ACL filters depending on the mathcing policy configured by
+:ts:cv:`proxy.config.url_remap.acl_matching_policy`.
+
+First Explicit Match Wins Policy
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I appreciate what you're trying to say here, but distinguishing the two behaviors as "explicit" and "implicit" is probably not helpful to the user. Nor do we want to say that the one is explicit, and the other is first match wins: if you think about it, both only apply to the first match. The real difference between these is what the match criteria is:

* In the new way added in 10, the ACL filters match based upon IP only, as they do for ip_allow.yaml.
* In the old way they match differently than ip_allow: they have to match both IP and method. 

I think this will be the most helpful framework for the user to understand and choose between the two policies. Thus I suggest something like this in these sections:

```
Match on IP and Method Policy
~~~~~~~~~~~~~~~~~~~~~~~

With this policy, ACL filters, named or inline, only take effect if both
IP and method match the incoming request. This policy is useful for
organizations that want ACL rules to additively allow or deny specific
methods in addition to other ACL filters and ip_allow.yaml rules.
Consider a filter like the following:

map http://www.example.com/ http://internal.example.com/ @action=deny @method=POST

The implicit @src_ip is all client IP addresses, so this filter will
match on any POST request matched by this remap rule from any client and
its action will be to deny such POST requests. For all other methods,
the filter will not take effect, thus allowing other active ACL filters
or an ip_allow.yaml rule to determine the action to take for any other
transaction.

...

Match on IP Policy 
~~~~~~~~~~~~~~

With this policy, ACL filters match solely based upon IP, meaning that
ACL filters match like ip_allow.yaml rules. This policy is useful for
organizations that want to have ACL filters behave like ip_allow rules
specific to remap targets. Thus, consider a filter like the following:

map http://www.example.com/ http://internal.example.com/ @action=allow @method=GET @method=HEAD

The implicit @src_ip is all client IP address, so this filter will apply
to all transactions matching this remap rule. Again, like an analogously
crafted ip_allow allow rule, this will allow ``GET`` and ``HEAD`` requests
while denying all other methods to the ``internal.example.com`` origin.
No other ACL filters or ip_allow rules will apply for this target.

This policy is new to ATS 10.

```

Reply:
Beside the naming of the policies, descriptions are clear. I'd use the same example to clarify the differences.

Reply:
After I considered naming for a while, I start calling them "Match on IP and Method Policy" and "Match on IP only Policy".